vpr/src/pack/greedy_clusterer.h

```diff
@@ -120,9 +145,72 @@ class GreedyClusterer {
                   Prepacker& prepacker,
                   bool allow_unrelated_clustering,
                   bool balance_block_type_utilization,
-                  AttractionInfo& attraction_groups);
+                  AttractionInfo& attraction_groups,
+                  DeviceContext& mutable_device_ctx);
 
 private:
+    /**
+     * @brief Given a seed molecule and a legalization strategy, tries to grow
+     *        a cluster greedily. Will return the ID of the cluster created.
```

Comment:
greedily, starting with the provided seed and adding whatever other molecules seem beneficial and legal.